internal/pkg/template/templates/workloads/services/static-site/cf.yml

```diff
@@ -95,8 +95,10 @@ Resources:
     Type: AWS::CloudFront::OriginAccessControl
     Properties:
       OriginAccessControlConfig:
-          Description: Access control for static s3 origin
-          Name: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-origin-access-control'
+          Description: !Sub 'Access control for static s3 origin for ${AppName}-${EnvName}-${WorkloadName}'
+          # Truncate the name to allow at most 64 characters.
+          {{$stackName := printf "%s-%s-%s" .AppName .EnvName .WorkloadName}}
+          Name: {{printf "%.64s" $stackName}}
```

Comment:
wait - i'm worried if this could happen
```
apppppppppp-envvvvvvvvvvvvvvvvvvvv-service1
apppppppppp-envvvvvvvvvvvvvvvvvvvv-service1
```
both would be truncated to `apppppppppp-envvvvvvvvvvv` 😓

Reply:
Well it is to help to _some extent_ but not every. I guess `apppppppppp-envvvvvvvvvvv` is relatively uncommon. We can deal with this problem completely by hashing but i don't want to sacrifice the readability either. In the future we could change if many ppl complain because they have really long app and env name.

Reply:
I think once truncate happens, it's likely end with only "app-env" left, because the "-service" part is always the last to truncate :( 

I think somewhere in the codebase that we equally distribute the quota to truncate among app, env and service, so it's more likely that the truncated name is unique.

Something like `subStringlength = (64-3) / 3`, and then return `app[len(app)-subStringlength:]-env[len(app)-subStringlength:]-svc[len(app)-subStringlength:]`

However I agree that people have workaround - worse case they could use override to hardcode a name. But this YAML patch file can never go away even after we update the code.

You make the call! I will approve.

Reply:
Or we truncate by 60 characters and append with 4 char hashing